Kernel/InterruptHandler.cpp

```diff
@@ -0,0 +1,53 @@
+#include <Kernel/InterruptHandler.h>
+#include <Kernel/SharedInterruptHandler.h>
+
+InterruptHandler::InterruptHandler(u8 irq)
+    : m_irq_number(irq)
+{
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).register_interrupt_handler(*this);
+    disable_irq();
+}
+
+InterruptHandler::~InterruptHandler()
+{
+    disable_irq();
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).unregister_interrupt_handler(*this);
+}
+
+bool InterruptHandler::is_enabled()
+{
+    return m_enabled;
+}
+
+void InterruptHandler::enable_irq()
+{
+    m_enabled = true;
+}
+
+void InterruptHandler::change_irq_number(u8 irq_number)
+{
+    bool was_enabled = m_enabled;
+    disable_irq();
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).unregister_interrupt_handler(*this);
+    m_irq_number = irq_number;
+    SharedInterruptHandler::get_shared_interrupt_handler(m_irq_number).register_interrupt_handler(*this);
+    if(was_enabled)
```

Comment:
Missing space after `if`, please use `clang-format`.

Reply:
My bad, I didn't use git-clang-format correctly... Shouldn't happen again since I configured my VS Code editor to use clang-format (style: none), so that works fine now :)